indico/modules/events/api.py

```diff
@@ -551,9 +550,10 @@ def _build_event_api_data(self, event):
             'modificationDate': self._serialize_date(event.as_legacy.getModificationDate()),
             'creationDate': self._serialize_date(event.created_dt),
             'creator': self._serialize_person(event.creator, person_type='Avatar', can_manage=can_manage),
-            'hasAnyProtection': event.as_legacy.hasAnyProtection(),
+            'hasAnyProtection': event.effective_protection_mode != ProtectionMode.public,
             'roomMapURL': event.room.map_url if event.room else None,
-            'visibility': Conversion.visibility(event.as_legacy),
+            # TODO
+            # 'visibility': Conversion.visibility(event.as_legacy),
```

Comment:
there's `event.visibility` now (not merged yet but very soon); same behavior as Category.visibility but with 0 for invisible being allowed